### PR TITLE
Bug 1232684 - Use makeFile to generate chrome.manifest from jar.mn

### PR DESCRIPTION
**Pull request by chrafuse**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,6 +145,8 @@ standalone: node_modules
 .PHONY: add-on
 add-on: node_modules
 	mkdir -p $(BUILT)/$@
+	sed -n -e '/^%/p' $@/jar.mn | sed -e "s/^% //g" | sed -e "s/%/chrome\//g" > $@/chrome.manifest
```
Comment (Standard8) [inline Makefile]: Can you separate this out to a separate rule please? No point in regenerating it each time - see some of the vendor library handling for instance.

Also, this should output direct to `$(BUILT)/add-on/chrome.manifest`, and we should git rm the existing`add-on/chrome.manifest`.

File: Makefile [hunk 2]
```diff
@@ -145,6 +145,8 @@ standalone: node_modules
...
+	sed -e '/^%/d' $@/jar.mn > $(BUILT)/$@/jar.mn
```
Comment (Standard8) [inline Makefile]: We don't need to generate the jar.mn here, that's just the source (we need it for when we export back to m-c).

File: Makefile [hunk 3]
```diff
@@ -145,7 +146,7 @@ standalone: node_modules
 .PHONY: add-on
 add-on: node_modules
 	mkdir -p $(BUILT)/$@
-	$(RSYNC) $@/chrome.manifest $@/chrome/bootstrap.js $(BUILT)/$@
+	$(RSYNC) $@/jar.mn $@/chrome/bootstrap.js $(BUILT)/$@
```
Comment (Standard8) [inline Makefile]: We don't need the jar.mn here - just remove it, then add `$(BUILT)/$(ADD-ON)/chrome.manifest` as a dependency on the `add-on:` line above.

File: Makefile [hunk 4]
```diff
@@ -164,6 +165,13 @@ add-on: node_modules
 	mkdir -p $(BUILT)/$@/chrome/locale/en-US
 	cat locale/en-US/$@.properties locale/en-US/shared.properties > $(BUILT)/$@/chrome/locale/en-US/loop.properties
 
+.PHONY: chrome-manifest
+chrome-manifest: $(ADD-ON)/jar.mn
```
Comment (Standard8) [inline Makefile]: No need for the .PHONY as this is a real dependency. We do however need to change this line to:

`$(BUILT)/$(ADD-ON)/chrome.manifest: $(ADD-ON)/jar,mn`

that sets up the output chrome.manifest as dependent on jar.mn.

File: add-on/chrome.manifest [hunk 5]
```diff
@@ -1,24 +0,0 @@
-content loop chrome/content/ contentaccessible=yes
-locale loop en-US chrome/locale/en-US/
```
Comment (Standard8) [inline add-on/chrome.manifest:2]: Please can you add a line to jar.mn to replace this. I'd forgotten about it before. You should just be able to add:

`% locale loop en-US %locale/en-US/`

If you could put that under the content line in add-on/jar.mn that'd be great

**Comment by Standard8** [comment]
Looks good, r=Standard8 with the additional comment fixed.

**Comment by chrafuse** [comment]
Added locale line from comments and pushed to mozilla master.

**Pull request closed without merging**